docs/src/main/sphinx/connector/elasticsearch.rst

```diff
@@ -119,6 +119,42 @@ This property is optional; the default is ``1m``.
 Enable or disable using the address published by Elasticsearch to connect for
 queries.
 
+``elasticsearch.union-schema-indices-for-alias``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Enable or disable merging of schemas of indices for governing the schema of an alias.
+If disabled, schema of the alias follows the schema of first index only.
+Same field with different data types in different indices for an alias casts
+as :ref:`raw JSON <elasticsearch-raw-json-transform>`.
+
+This property is optional; the default value is ``false``.
+
+``elasticsearch.fail-on-alias-schema-mismatch``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Enable or disable to fail the query if same field from different indices, for an alias, have different data types.
+
+This property is optional; the default value is ``false`` i.e.,
+casts mismatch column as :ref:`raw JSON <elasticsearch-raw-json-transform>`.
+
+``elasticsearch.datatype-for-alias-schema-mismatch``
```

Comment:
I think we should avoid exposing this and just make the default behavior to cast mismatched fields to raw json. If you don't like this behavior, turn on the `elasticsearch.fail-on-alias-schema-mismatch` property. Curious to hear @martint's thoughts here.

Reply:
Yeah makes sense. More configs lead to more confusion. @martint Thoughts?

Reply:
We may also want to consider doing something smarter if the types being merged are compatible with each other. For instance, when merging `long` with `integer`, we may want to present the wider type instead of falling back to json.

Reply:
Done